### DISCUSSION TITLE
Idea: Extension for extension devs

### DISCUSSION THREAD

**Discussion by worksofliam** [microsoft/vscode-discussions#518] [Extension Development QnA]
So after a while, when your extension is huge, managing the `package.json` is a real pain. I am thinking of creating a simple extension which makes managing your extension `package.json` simpler. I've been thinking about this for a few weeks, but here's a simple overview of how this tree view might work:

Single view:

* Possible/used context values
* All commands
   * Show where it’s used
   * Keybindings for command
   * Enablement
* View containers
   * Grouped by activityBar and panel
   * Id
   * Title
   * Icon
   * Views below it
      * Welcome text (viewsWelcome)
      * Type (tree/webview)
      * id
      * name
      * Contextual title
      * When clause
      * Default visibility
      * Title buttons (as determined by view == viewId)
         * When clause
      * TreeItem buttons (as determined by view == viewId)
         * Perhaps group by viewItem context value?
         * When clause
* Editor commands
   * Title (tab) (menu.editor/title)
   * Editor (editor/context)

Functions to:

* Create new view container
* Create new view below container
   * Add commands to view title/items
   * Set the default welcome text
   * Perhaps generate a base class for this?
* Register new commands
    * Set keybindings
* Add commands to editor title/context
* Way more

Most of the wizards could be powered by some simple webview powered forms. There is a lot more it could do, but the main is to make maintaining the larger `package.json` files easier.

**Comment by digitarald** [comment]
I wonder how well Copilot could work to add the right code to different files. I can see how some guidance could make it faster to generate nice JSON and the JS/TS scaffold.

**Comment by burkeholland** [comment]
Closing here because this is more than 30 days old. Would love to hear about the solution you came up with. Please feel free to re-open and provide the answer.
